### PR TITLE
Update metrics documentation for Cloud Monitoring/Stackdriver

### LINKED ISSUE #2850: Docs: Rename "Stackdriver" to "Cloud Monitoring"
See: https://agones.dev/site/docs/guides/metrics/

It still uses "Stackdriver" in the docs, which is poor SEO for anyone looking for "Cloud Monitoring".

Suggestion: 

Change the headings to "Cloud Monitoring (formerly Stackdriver)" to cover all bases - and also explain why the Helm flag to enable Cloud Monitoring is still `agones.metrics.stackdriverEnabled`

### PR DESCRIPTION
**Pull request by junninho**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
- Update Stackdriver references to Cloud Monitoring
- Include missing steps to get Cloud Monitoring up and running
- Add step to resolution for the error mentioned in the troubleshooting section

**Which issue(s) this PR fixes**:
Closes #2850 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like your docs change have introduced a broken link unfortunately:

```
site/docs/guides/metrics/index.html
  hash does not exist --- site/docs/guides/metrics/index.html --> #stackdriver-installation
```

**Comment by roberthbailey** [comment]
Looks like an error was caught by htmltest:

```
htmltest started at 06:05:57 on /tmp/website
========================================================================
site/docs/guides/metrics/index.html
  hash does not exist --- site/docs/guides/metrics/index.html --> #stackdriver-installation
```

File: site/content/en/docs/Guides/metrics.md [hunk 1]
```diff
@@ -207,24 +207,48 @@ kubectl port-forward deployments/grafana 3000 -n metrics
...
+Before proceeding, ensure you have created a metrics node pool as mentioned in the Google Cloud [installation guide]({{< ref "/docs/Installation/Creating Cluster/gke.md" >}}).
+
+The default metrics exporter installed with Agones is Prometheus. If you are using the [Helm installation]({{< ref "/docs/Installation/Install Agones/helm.md" >}}), you can install or upgrade Agones to use Cloud Monitoring, using the following chart parameters:
 ```bash
 helm upgrade --install --wait --set agones.metrics.stackdriverEnabled=true --set agones.metrics.prometheusEnabled=false --set agones.metrics.prometheusServiceDiscovery=false my-release-name agones/agones --namespace=agones-system
 ```
 
-With this configuration only Stackdriver exporter would be used instead of Prometheus exporter.
+If you are using the [YAML installation]({{< ref "/docs/Installation/Install Agones/yaml.md" >}}), you will need to update the necessary parameters in the install.yaml file using the following commands:
```
Comment (aimuz) [inline site/content/en/docs/Guides/metrics.md:231]:
```suggestion
If you are using the [YAML installation]({{< ref "/docs/Installation/Install Agones/yaml.md" >}}), you will need to update the necessary parameters in the `install.yaml` file using the following commands:
```

**Review by aimuz — COMMENTED** [review]
Looks good

File: site/content/en/docs/Guides/metrics.md (hunk 1 above)
Comment (aimuz) [reply site/content/en/docs/Guides/metrics.md:231]: This seems to be a little better, don't you think?

**Comment by aimuz** [comment]
LGTM

Thank you

File: site/content/en/docs/Guides/metrics.md [hunk 2]
```diff
@@ -207,24 +207,48 @@ kubectl port-forward deployments/grafana 3000 -n metrics
...
+  --set agones.metrics.stackdriverEnabled=true \
+  --set agones.metrics.prometheusEnabled=false \
+  --set agones.metrics.prometheusServiceDiscovery=false \
+  > install-custom.yaml
+```
 
-If you would like to enable stackdriver in conjunction with [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity), there are a few extra steps you need to follow:
+```bash
+kubectl apply -f install-custom.yaml
+```
```
Comment (markmandel) [inline site/content/en/docs/Guides/metrics.md:245]: 🤔 I'm not sure about this section. I'm kinda figuring it's not necessary and possibly confusing.

If you are using the YAML install, running this command doesn't actually change the install.yaml that is hosted on github for the release - it just _gives you_ an yaml file. 

If someone is familiar with Helm and wants a yaml file instead, I would argue it's probably likely they already know how to do this.

If they aren't familiar with Helm, I'd worry this section is confusing. I'd be tempted to drop it.

**Review by markmandel — COMMENTED** [review]
Thank you for doing this work! This is great. I am worried that section on outputting a yaml file could be confusing though - have a look at my comment and let me know what you think.

File: site/content/en/docs/Guides/metrics.md (hunk 2 above)
Comment (junninho) [reply site/content/en/docs/Guides/metrics.md:245]: Hmm I definitely could have phrased that more accurately. Similarly to the Install Agones using YAML page, I do think having some guidance around what to do if you are using YAML would be helpful. An alternative would be to simply add a note linking to [this part](https://agones.dev/site/docs/installation/install-agones/yaml/#:~:text=Copy-,To,-change%20the%20configurable) of the page to change the parameters.

I'm also not opposed to dropping it but let me know what you think.

Comment (markmandel) [reply site/content/en/docs/Guides/metrics.md:245]: Personally, I don't see a lot of people finding confusion with generating yaml from Helm -- but it's entirely possible that's a blind spot I have.

> An alternative would be to simply add a note linking to [this part](https://agones.dev/site/docs/installation/install-agones/yaml/#:~:text=Copy-,To,-change%20the%20configurable) of the page to change the parameters.

Seems like a good compromise to me 👍🏻  Could even do it as an "Note" box if you want to highlight it, e.g.

https://github.com/googleforgames/agones/blob/1cc9a3bc0afc8e7dc98810b1f6ec09be2aa59e72/site/content/en/docs/Guides/metrics.md?plain=1#L114-L116

**Pull request merged**